compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/repro-undefined-expression-of-jsxexpressioncontainer.expect.md

```diff
@@ -48,32 +48,26 @@ import { c as _c } from "react/compiler-runtime";
 import { StaticText1, Stringify, Text } from "shared-runtime";
 
 function Component(props) {
-  const $ = _c(6);
+  const $ = _c(4);
   const { buttons } = props;
-  let nonPrimaryButtons;
-  if ($[0] !== buttons) {
-    [, ...nonPrimaryButtons] = buttons;
-    $[0] = buttons;
-    $[1] = nonPrimaryButtons;
-  } else {
-    nonPrimaryButtons = $[1];
-  }
   let t0;
-  if ($[2] !== nonPrimaryButtons) {
+  if ($[0] !== buttons) {
+    const [, ...nonPrimaryButtons] = buttons;
+
     t0 = nonPrimaryButtons.map(_temp);
-    $[2] = nonPrimaryButtons;
-    $[3] = t0;
+    $[0] = buttons;
```

Comment:
this is the example where i realized we weren't inferring array pattern spread as an array type

Reply:
Ahh makes sense. `MergeReactiveScopesThatInvalidateTogether` uses type info as a proxy for whether something was newly created, which currently works due to how we infer types